### PR TITLE
Correction-matrix-StandingWave

### PR DESCRIPTION
**Pull request by SPHYaru**
1. add the weighted kernel gradient correction.
2. add the standing wave case.

### CODE REVIEW

**Comment by Xiangyu-Hu** [comment]
> 1. add the weighted kernel gradient correction.
> 2. add the standing wave case.

Hi, according to the file changed. It seems quite some changes in the library. Could you move these file to the folder of the test?

**Comment by Xiangyu-Hu** [comment]
I seems that CI has not pass. Could you recompile it with update master branch?

**Comment by ChiZhangatTUM** [comment]
Why un-used variable is reported as error?

**Review by DongWuTUM — APPROVED** [review]

File: tests/2d_examples/test_2d_standing_wave/fluid_dynamics_complex_correct.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+namespace SPH
+{
+	//=====================================================================================================//
+	namespace fluid_dynamics
+	{
+		//=================================================================================================//
+		template <class BaseIntegration1stHalfCorrectType>
+		void BaseIntegration1stHalfCorrectWithWall<BaseIntegration1stHalfCorrectType>::interaction(size_t index_i, Real dt)
+		{
+			BaseIntegration1stHalfCorrectType::interaction(index_i, dt);
```
Comment (Xiangyu-Hu) [inline tests/2d_examples/test_2d_standing_wave/fluid_dynamics_complex_correct.hpp:36]: This is the same as before. Do we really need a new class?

File: tests/2d_examples/test_2d_standing_wave/fluid_dynamics_inner_correct.hpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			rho_[index_i] += drho_dt_[index_i] * dt * 0.5;
+			p_[index_i] = fluid_.getPressure(rho_[index_i]);
+			pos_[index_i] += vel_[index_i] * dt * 0.5;
+			p_B_[index_i] = p_[index_i] * B_[index_i];
+		}
+		//=================================================================================================//
+		template <class RiemannSolverType>
+		void BaseIntegration1stHalfCorrect<RiemannSolverType>::update(size_t index_i, Real dt)
+		{
+			vel_[index_i] += (acc_prior_[index_i] + acc_[index_i]) * dt;
```
Comment (Xiangyu-Hu) [inline tests/2d_examples/test_2d_standing_wave/fluid_dynamics_inner_correct.hpp:53]: This also the same as the one in the library. Do wee need a repeated new class?

**Review by Xiangyu-Hu — CHANGES_REQUESTED** [review]

File: tests/2d_examples/test_2d_standing_wave/fluid_dynamics_inner_correct.hpp (hunk 2 above)
Comment (Xiangyu-Hu) [reply tests/2d_examples/test_2d_standing_wave/fluid_dynamics_inner_correct.hpp:53]: The code quality seems should be improved.

**Pull request closed without merging**